src/Features/CSharp/Portable/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -227,7 +227,8 @@ private static async Task HandleSingleTypeAsync(CompletionContext context, Seman
                         continue;
                     }
 
-                    if (!SymbolEqualityComparer.Default.Equals(type, symbolType)
+                    // We only show static properties/fields of compatible type if containing type is NOT marked with completionlist tag.
+                    if (!isEnumOrCompletionListType && !SymbolEqualityComparer.Default.Equals(type, symbolType)
                         || !staticSymbol.IsAccessibleWithin(enclosingNamedType)
                         || !staticSymbol.IsEditorBrowsable(hideAdvancedMembers, semanticModel.Compilation))
```

Comment:
tbh i'm not sure how to understand the logic here.  there's a lot of && and ! and || mixed together.  Can you break this out?

Reply:
done